.github/workflows/src/issues.js

```diff
@@ -0,0 +1,52 @@
+/**
+ * Retrieves the PR number associated with a specific commit SHA
+ * @param {Object} params
+ * @param {String} params.head_sha - The head_sha
+ * @param {typeof import("@actions/core")} params.core - GitHub Actions core for logging
+ * @param {import("@octokit/core").Octokit & import("@octokit/plugin-rest-endpoint-methods/dist-types/types.js").Api} params.github - GitHub API client
+ * @returns {Promise<{issueNumber: number}>} - The PR number or NaN if not found
+ */
+export async function getIssueNumber({ head_sha, core, github }) {
```

Comment:
Could this be moved into context.js, and returned automatically as the value of "output.issue_number" instead of NaN?

Reply:
As talked offline, the reason is that we want to save the call of searching issues if there are no label action to make.